### PR TITLE
Tests: more tests for selector-type-no-unknown (#2336).

### PR DESCRIPTION
**Pull request by evilebottnawi**
> Which issue, if any, is this issue related to?
#2336

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/selector-type-no-unknown/__tests__/index.js [hunk 1]
```diff
@@ -50,6 +50,16 @@ testRule(rule, {
   }, {
     code: "a:nth-child(n) {}",
   }, {
+    code: "input:not(*) {}",
```
Comment (hudochenkov) [inline lib/rules/selector-type-no-unknown/__tests__/index.js]: I'm not sure what elements will be targeted by this selector. Do you have idea? :) It targets `input`, which should not be targeted by `*`. But every `input` is targeted by `*`.

File: lib/rules/selector-type-no-unknown/index.js [hunk 2]
```diff
@@ -85,7 +85,9 @@ const rule = function (actual, options) {
           const tagName = tagNode.value
           const tagNameLowerCase = tagName.toLowerCase()
 
-          if (htmlTags.indexOf(tagNameLowerCase) !== -1 || svgTags.indexOf(tagNameLowerCase) !== -1 || nonStandardHtmlTags.has(tagNameLowerCase)) {
+          if (htmlTags.indexOf(tagNameLowerCase) !== -1 || svgTags.indexOf(tagNameLowerCase) !== -1
```
Comment (hudochenkov) [inline lib/rules/selector-type-no-unknown/index.js]: I think this change is unrelated to this PR.

Comment (evilebottnawi) [inline lib/rules/selector-type-no-unknown/index.js]: @hudochenkov i think we should change the length of the string to 120 (щк 80), it's just not possible to read, maybe it should be done in another PR.

File: lib/rules/selector-type-no-unknown/__tests__/index.js (hunk 1 above)
Comment (evilebottnawi) [inline lib/rules/selector-type-no-unknown/__tests__/index.js]: @hudochenkov see first note https://developer.mozilla.org/ru/docs/Web/CSS/:not :smile: it is possible
